File: android/app/src/main/java/com/aitutor/app/domain/repository/WritingRepository.kt
package com.aitutor.app.domain.repository

import com.aitutor.app.domain.model.WritingEvaluation
import com.aitutor.app.domain.model.WritingTaskType
import com.aitutor.app.domain.model.WritingTopic

interface WritingRepository {
    fun randomTopic(taskType: WritingTaskType): WritingTopic

    suspend fun evaluate(topic: WritingTopic, essay: String): Result<WritingEvaluation>

    suspend fun saveEssay(
        taskType: WritingTaskType,
        topic: WritingTopic,
        essay: String,
        evaluation: WritingEvaluation
    ): Result<Unit>
}

File: ios/AITutor/Domain/Repository/WritingRepository.swift
import Foundation

protocol WritingRepository: AnyObject {
    func randomTopic(taskType: WritingTaskType) -> WritingTopic

    func evaluate(topic: WritingTopic, essay: String) async throws -> WritingEvaluation

    func saveEssay(
        taskType: WritingTaskType,
        topic: WritingTopic,
        essay: String,
        evaluation: WritingEvaluation
    ) async throws
}
